dbt_platform_helper/utils/files.py

```diff
@@ -102,3 +106,76 @@ def combine_env_data(data):
     enriched_config["environments"] = defaulted_envs
 
     return enriched_config
+
+
+def read_supported_versions_from_cache(resource_name):
+
+    platform_helper_config = read_file_as_yaml(platform_helper_config_file)
+
+    return platform_helper_config.get(resource_name).get("versions")
+
+
+def write_to_cache(resource_name, supported_versions):
+
+    platform_helper_config = {}
+
+    if os.path.exists(platform_helper_config_file):
+        platform_helper_config = read_file_as_yaml(platform_helper_config_file)
+
+    cache_dict = {
+        resource_name: {
+            "versions": supported_versions,
+            "date-retrieved": datetime.now().strftime("%d-%m-%y %H:%M:%S"),
+        }
+    }
+
+    platform_helper_config.update(cache_dict)
+
+    with open(platform_helper_config_file, "w") as file:
+        file.write("# [!] This file is autogenerated via the platform-helper. Do not edit.\n")
+        yaml.dump(platform_helper_config, file)
+
+
+def cache_refresh_required(resource_name):
+    """
+    Checks if the platform-helper should reach out to AWS to 'refresh' its
+    cached values.
+
+    An API call is needed if any of the following conditions are met:
+        1. No cache file (.platform-helper-config.yml) exists.
+        2. The resource name (e.g. redis, opensearch) does not exist within the cache file.
+        3. The date-retrieved value of the cached data is > than a time interval. In this case 1 day.
+    """
+
+    api_call_required = True
+
+    if os.path.exists(platform_helper_config_file):
+
+        platform_helper_config = read_file_as_yaml(platform_helper_config_file)
+
+        if platform_helper_config.get(resource_name):
+
+            api_call_required = check_if_cached_datetime_is_greater_than_interval(
```

Comment:
By using guard clauses and reducing the nesting, we wouldn't need to store the result at all, and could instead return it directly